external/sql/storm-sql-runtime/src/test/org/apache/storm/sql/TestUtils.java

```diff
@@ -60,6 +62,25 @@ public static String result(String accumulator) {
   }
 
 
+  public static class TopN {
+    public static PriorityQueue<Integer> init() {
+      return new PriorityQueue<>();
+    }
+    public static PriorityQueue<Integer> add(PriorityQueue<Integer> accumulator, Integer n, Integer val) {
+      if (accumulator.size() >= n) {
```

Comment:
One missed spot is that it should do nothing if n is 0. Now if n is 0, it will behave same as top 1. n should be greater than 0 anyway but we can do some defensive programming.
Other then that it looks good.